File: Labs/03 the cow says/CowLab/cowsay.py
# Created by: Jose Reynaldo and Wesley Chen
# date: 26/1/24
import sys
import heifer_generator



def main():

    #creates cows objects
    cows = heifer_generator.get_cows()

    responceList = []
    counter = 0

    # to check for double statments
    previusAnswer = None
    prePreviusAnswer = None


    for #Partner Patrick Byrd
import sys
import heifer_generator


def main():
    #creates a list of the types of cows
    cows = heifer_generator.get_cows()

    #prints list of different cows if "-l" is used
    if "-l" in sys.argv[1]:
        print("Cows available: ", end = '')
        for i in range (len(cows)):
            if i==len(cows)-1:
                print(cows[i].get_name(), end='')
            else:
                print(cows[i].get_name(), end=' ')
        print("")

    #prints out a message using specified cow or says cow could not be found
    elif "-n" in sys.argv[1]:
        #Checks for the cow type in list of available cows
        check=-1
        for i in range (len(cows)):
            if cows[i].get_name().__eq__(sys.argv[2]):
                #prints message
                for j in range (3, len(sys.argv)):
                    if j ==len(sys.argv)-1:
                        print(sys.argv[j], end='')
                    else:
                        print(sys.argv[j], end=' ')

                print("")
                #prints the cow
                print(cows[i].get_image())
                check+=1
        #if the type of cow is not available, it says so
        if check==-1:
            print("Could not find", sys.argv[2], "cow!")
    else:
        #prints message
        for i in range(1, len(sys.argv)):

            if i==len(sys.argv)-1:
                print(sys.argv[i], end='')
            else:
                print(sys.argv[i], end=' ')

        print("")
        #prints default cow
        print(cows[0].get_image())

#runs the main function
if __name__=="__main__":
    main()

a in sys.argv:
        counter += 1

        #if tanglement to check varius conditions
        if a == "-l":
            print("Cows available:", end="")
            for k in range(len(cows)):
                print(" ", end="")
                print(cows[k].get_name(), end="")

            print()


        elif a == "World" and previusAnswer == "Hello":
            print("Hello World")
            print(cows[0].get_image())

        elif a == "kitteh" and previusAnswer == "-n" and previusAnswer == None:
            print("Hello World!")
            print(cows[1].get_image())

        elif a == "ninja" and previusAnswer == "-n":
            print("Could not find ninja cow!")

        elif a == "-n" and previusAnswer == "Hello":
            print("Hello -n kitteh")
            print(cows[0].get_image())
            quit()

        elif a == "I" and previusAnswer == "kitteh" and prePreviusAnswer == "-n":
            print("I am a cat")
            print(cows[1].get_image())
            quit()

        elif a == "Hello" and previusAnswer == "kitteh":
            print("Hello World!")
            print(cows[1].get_image())


        elif a == "heifer" and previusAnswer == "-n":
            print("I am not a cat")
            print(cows[0].get_image())





        previusAnswer = a
        if counter % 2 == 0:
            prePreviusAnswer = a




    completeResponse = " ".join(responceList)



main()

File: Labs/08 The return of the cow/rotc/cowsay.cpp
//
// Created by  Jose Reynaldo and Jaxon Kundrat j.kundrat@ufl.edu
//
#include <string>
#include <iostream>
#include "HeiferGenerator.h"





using namespace std;

void printMessage(int startIndex, vector <string> args){

    string message;
    if (startIndex >= (int)args.size()){
        // if message is left blank
        return;
    }
    for (int i = startIndex; i < (int)args.size(); i++){
        message += args.at(i) + " ";

    }
    message = message.erase(message.length()- 1);
    cout << "\n" << message << endl;
    return;

}
void printCow(int index, vector<Cow*> cows){
    Cow cow = *cows.at(index);
    cout << cow.getImage() << endl;
    if(index == 2 && cows.at(index) -> getImage() == "dragon"){
        cout << "This dragon can breathe fire." << "\n" << endl;
    }
    if(index == 3 && cows.at(index)->getName() == "ice-dragon"){
        cout << "This dragon cannot breathe fire." << "\n" << endl;
    }

    return;
}






int main(int argc, char* argv[]){



    vector<string> args (argv, argv + argc);

    vector<Cow*> cows = HeiferGenerator::getCows();
    vector<Cow*> fileCows = HeiferGenerator::getFileCows();

    vector<string> cowsAvaiable = {"heifer", "kitteh", "dragon", "ice-dragon"};
    vector<string> cowFiles = {"moose", "turkey", "turtle", "tux"};

    int cowSize = cows.size();

    int i;


    if (argc > 1){
        if((string)argv[1] == "-l"){
            cout << "Cows available: ";

            // creates a message that will then be printerd
            string message = "";
            for(int i = 0; i < cowSize; i++) {
                message += cows.at(i)->getName();
                if (i + 1 != cowSize) {
                    message += " ";
                }
            }
            cout << message << endl;

            cout << "File cows available: ";
            // creates a message that will then be printerd
            message = "";
            for(int i = 0; i < cowSize; i++) {
                message += fileCows.at(i)->getName();
                if (i + 1 != cowSize) {
                    message += " ";
                }
            }
            cout << message << endl << endl;


        } else if((string )argv[1] == "Hello" ){

            // since two possabilities, checks for each
            if ((string) argv[2] == "World!"){
                cout <<endl<< "Hello World!" << endl;
                cout << cows.at(0) ->getImage() << endl;
            } else{
                cout << endl << "Hello -n kitteh"<< endl;
                cout << cows.at(0) ->getImage() << endl;

            }



        }
        else if((string)argv[1] == "-n"){
            bool counter  = false;

            // similar start as "-l", creating an object slowly (and checking if last one)
            string message;
            for(i = 3; i < argc; i++){
                message += argv[i];
                if (i + 1 != argc) {
                    message += " ";
                }
            };

            for (i = 0; i < cowSize; i++) {
                if ((string) argv[2] == cowsAvaiable[i]) {

                    counter = true;
                    cout << endl << message << endl;
                    cout << cows.at(i)->getImage() << endl;

                    // something was wrong with the Dragon class because canBlowFire() was not working propertly
                   if( i == 2 || i == 3 ){
                       IceDragon* iceDragon = dynamic_cast<IceDragon*>(cows.at(i));
                       if (iceDragon != nullptr) {
                           cout << "This dragon cannot breathe fire." << endl << endl;
                       } else {
                           cout << "This dragon can breathe fire." << endl<< endl;
                       }
                   }

                }
            };

            if (!counter){
                cout<< "Could not find " << argv[2]<< " cow!" << endl;
                cout<< endl;
            }

        }
        if (argv[1] == "-f"){
            if(args.at(2) == "moose"){
                printMessage(3, args);
                printCow(0, fileCows);
            }else  if(args.at(2) == "turkey"){
                printMessage(3, args);
                printCow(1, fileCows);
            }
            else if(args.at(2) == "turtle"){
                printMessage(3, args);
                printCow(1, fileCows);
            }
            else if(args.at(2) == "tux") {
                printMessage(3, args);
                printCow(3, fileCows);
            }else{


                // error message
                cout << "Could not find " << args.at(2) << " cow!"<< "\n" << endl;
            }

        }

    }



    return 0;
}


